src/dist/manifestation.rs

```diff
@@ -12,6 +12,33 @@ use crate::dist::prefix::InstallPrefix;
 use crate::dist::temp;
 use crate::utils::utils;
 use std::path::Path;
+use std::thread;
+use std::time::{Duration, SystemTime};
+
+macro_rules! retry_with_backoff {
+    ( $( $call:expr )+ ) => {
+        {
+            let mut ret;
+            let mut retries = 3;
```

Comment:
Might be good to be able to configure this somewhere or at least for it to be a const somewhere obvious.

Reply:
I filled #1667 and I'd like this to be configurable. Our main use case is CI, and the number of retries that might make sense for Travis-CI is probably different than for Appveyor, Cirrust-CI, Azure, etc. 

Also, 3 retries weren't enough for travis, and now we are using 5 retries: https://github.com/rust-lang/libc/blob/master/ci/build.sh#L31 

The backoff time of our retries includes the time it takes for rustup to fail after a download fails, so it might be slightly different from what's done here (but larger I expect), so we'd might need even a larger number of retries here.